main/src/main/scala/sbt/Defaults.scala

```diff
@@ -739,7 +743,17 @@ object Defaults extends BuildCommon {
   def runnerInit: Initialize[Task[ScalaRun]] = Def.task {
     val tmp = taskTemporaryDirectory.value
     val si = scalaInstance.value
-    if (fork.value) new ForkRun(forkOptions.value) else new Run(si, trapExit.value, tmp)
+    val s = streams.value
+    val options = javaOptions.value
+    if (fork.value) {
+      s.log.debug(s"javaOptions: $options")
+      new ForkRun(forkOptions.value)
+    } else {
+      if (options.nonEmpty) {
+        s.log.warn("javaOptions will be ignored, fork is set to false")
```

Comment:
I'm still missing some info to get the scoped task, but that's where I am right now :

```
[warn] */*:javaOptions will be ignored, */*:fork is set to false
```